vignettes/vcr.Rmd

```diff
@@ -1,191 +1,172 @@
+Sometimes it also makes sense to manually edit the cassettes.
+Manually editing is typically most useful when you're testing error handling code, because it allows you to create responses that would otherwise be hard to get the API to generate.
 
-## Terminology
-* _vcr_: the name comes from the idea that we want to record something and play it back later, like a vcr
-* _cassette_: A _thing_ to record HTTP interactions to. Right now the only option is the file system (writing to files), but in the future could be other things, e.g. a key-value store like Redis
-* _fixture_: A fixture is something used to consistently test a piece of software. In this case, a cassette (just defined above) is a fixture - used in unit tests. If you use our setup function `vcr_setup()` the default directory created to hold cassettes is called `fixtures/` as a signal as to what the folder contains.
-* Persisters: how to save requests - currently only option is the file system
-* _serialize_: translating data into a format that can be stored; here, translate HTTP request and response data into a representation on disk to read back later
-* Serializers: how to serialize the HTTP response - currently only option is YAML; other options in the future could include e.g. JSON
-* _insert cassette_: create a cassette (all HTTP interactions will be recorded to this cassette)
-* _eject cassette_: eject the cassette (no longer recording to that cassette)
-* _replay_: refers to using a cached result of an http request that was recorded earlier
+## Request matching
 
-## Missing features
+By default, vcr looks for matching requests using just the HTTP method and the URI.
+If you need to match requests differently you can use `match_requests_on` parameter.
+The most likely reason to change this is to also match on the request body.
+You can do this in two ways: with `"body"` or with `"body_json"`.
+If your API uses JSON (like most modern APIs) you should use the `body_json` request matcher: that will parse the body as JSON so you'll get more informative messages if a new request doesn't match a request saved in a cassette.
 
-There's a number of features in this package that are not yet supported, but for which their parameters are found in the package. 
+Here's a little example with logging turned on, so you can see exactly what's happening:
 
-We've tried to make sure the parameters that are ignored are marked as such. Keep an eye out for package updates for changes in these parameters, and/or let us know you want it and we can move it up in the priority list.
+```{r}
+#| label: body-record
 
-## Examples
+get_data <- function(b = 1) {
```

Comment:
Should this be in a different chunk to clarify it's in a script under `R/`?
If the vignette is switched to `qmd` could we use the filename attribute? https://quarto.org/docs/output-formats/html-code.html#code-filename

Reply:
I'll add some comments; I don't want to switch to quarto just for this.